File: 328/2A.rb
a = Array.new(8)
c = [0] * 8
for x in 0..7
    a[x] = gets
end
flag = 0
for x in 0..7
    for y in 0..7
        if a[x][y] == "W" and c[y] == 0
            w = x;
            flag = 1;
            break;
        end
        if a[x][y] == "B"
            c[y] = 1;
        end
    end
    if flag == 1
        break
    end
end
c = [0] * 8
flag = 0
for x in 0..7
    for y in 0..7
        if a[7 - x][y] == 'B' and c[y] == 0
            b = x;
            flag = 1;
            break;
        end
        if a[7 - x][y] == 'W'
            c[y] = 1;
        end
    end
    if flag == 1
        break;
    end
end
if w > b
    print "B\n"
else
    print "A\n"
end

File: 360/2A.cc
#include<bits/stdc++.h>

using namespace std;

bool mat[110][110];
int n, d;
char ch;

int main() {
    int ans = 0;
    bool flag;
    int sum;

    cin >> n >> d;
    sum = 0;
    for(int i = 0;i < d;i ++) {
        flag = 1;
        for(int j = 0;j < n;j ++) {
            cin >> ch;
            while (ch != '0' && ch != '1') cin >> ch;
            if(ch == '0') flag = 0;
        }
        if(flag) {
            ans = max(ans, sum);
            sum = 0;
        } else sum ++;
    }
    ans = max(ans, sum);
    cout << ans << endl;

    return 0;
}
